ql/src/java/org/apache/hadoop/hive/ql/exec/vector/expressions/TruncDateFromTimestamp.java

```diff
@@ -147,18 +148,29 @@ protected void truncDate(ColumnVector inV, BytesColumnVector outV, int i) {
   }
 
   protected void processDate(BytesColumnVector outV, int i, Date date) {
-    if ("MONTH".equals(fmt) || "MON".equals(fmt) || "MM".equals(fmt)) {
+    switch (fmt) {
+    case "YEAR":
+    case "YYYY":
+    case "YY":
+      date.setMonth(1);
```

Comment:
date.setMonth(1) might be missing here?

Reply:
@pgaref Thanks so much for taking a look.  I'm not sure however what your comment means.

The year stuff get `Month=1` and `dayOfMonth=1`.  The Month stuff gets only `dayOfMonth=1`.

This switch statement "falls through" to cover both outcomes.

Reply:
Hey @belugabehr  -- it was actually the fall through that confused me, I see what you are doing now
In the original call it was part of an else if statement taking care of both.